ibm_cloud_sdk_core/base_service.py

```diff
@@ -309,6 +319,17 @@ def prepare_request(self,
 
         self.authenticator.authenticate(request)
 
+        # Compress the request body if applicable
+        current_encoding = request['headers']
```

Comment:
I got a little confused by the header handling code in this function, until I expanded my view to see where "headers" was initialized above.   My suggestion would be use the local variable "headers" exclusively as the dict containing the current set of request headers, until it's time to simply copy them to `request['headers']` near the end of prepare_request().

Reply:
Right, I'm not sure why I needed to make a separate local variable for this, so I'm just the existing headers now